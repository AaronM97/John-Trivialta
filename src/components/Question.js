import React from 'react'

export default function Question(props) {

  let correct = {isCorrect : true, answer : props.question.correct_answer}
  let incorrect = props.question.incorrect_answers.map(answer => ({ isCorrect : false , answer : answer }))
  
  function shuffle (array) {
    return array.sort(() => Math.random() - 0.5)
  }  

  // putting answers into to a new single answers array with a spread operator as objects so we can identify which is correct
  let answers = shuffle([correct, ...incorrect])

  function checkAnswer (e) {
    console.log(e.target.id)
  }

  return (
    <div>
      <p>{props.question.question}</p>

      {answers.map((answer, index) => (
        <p id={answer.isCorrect.toString()} onClick={checkAnswer} key={index}>{answer.answer}</p>
      ))}
      {/* <p>{props.question.correct_answer}</p>
      <p>{props.question.incorrect_answers[0]}</p>
      <p>{props.question.incorrect_answers[1]}</p>
      <p>{props.question.incorrect_answers[2]}</p> */}
    </div>
  )
}
