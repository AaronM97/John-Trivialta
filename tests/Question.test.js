import React from 'react';
import ReactDOM from 'react-dom';
import Question from '../src/components/Question';
import Enzyme, { shallow } from 'enzyme';
import { expect as exp } from 'chai';

describe('renders Question without crashing', () => {
    it('smoke test', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Question />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('snapchot smoke test', () => {
        exp(shallow(<Question />)).toMatchSnapshot();
    })
})

