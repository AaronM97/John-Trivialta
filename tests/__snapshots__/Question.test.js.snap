// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Question without crashing should render correctly 1`] = `
<div>
  <p
    className="questionCounter"
  >
    You are on Question 
    2
     / 10
  </p>
  <p
    className="questionQuestion"
  >
    On a dartboard, what number is directly opposite No. 1?
  </p>
  <Answer
    background="white"
    clicker={[MockFunction]}
    hasbeenclicked={false}
    isCorrect="true"
    key="0"
    text="19"
  />
  <Answer
    background="white"
    clicker={[MockFunction]}
    hasbeenclicked={false}
    isCorrect="false"
    key="1"
    text="15"
  />
  <Answer
    background="white"
    clicker={[MockFunction]}
    hasbeenclicked={false}
    isCorrect="false"
    key="2"
    text="12"
  />
  <Answer
    background="white"
    clicker={[MockFunction]}
    hasbeenclicked={false}
    isCorrect="false"
    key="3"
    text="20"
  />
  <button
    onClick={[MockFunction]}
  >
    Skip
  </button>
  <p>
    If you skip this question you get no score for it :(
  </p>
</div>
`;
